documentation/intermediate-ca.md

```diff
@@ -0,0 +1,199 @@
+      end
+    end
+    Puppet::Application::GenCACSR.new.run
+
+Use this CSR to generate the intermediate CA certificate that Puppet will be using.
+
+### Submit the generated CSR for signing
+
+The signing procedure will depend on the PKI that your organization is deployed; consult your PKI administrators for more information.
+
+### Construct the CA certificate chain
+
+Puppet Server needs to present the full certificate chain to clients in order for the client to authenticate the server. The certificate chain is constructed by concatenating the CA certificates together, starting with the new intermediate CA certificate and descending to the root CA certificate.
+
+    cat puppet-intermediate-certificate.pem \
+        org-intermediate-certificate.pem \
+         org-root-certificate.pem > ca-bundle.pem
+
+In this example, `puppet-intermediate-certificate.pem` is signed CA certificate that you'll be using with your Puppet PKI, and will be given to you by your PKI administrators. The `org-intermediate-certificate.pem` and `org-root-certificate.pem` are the CA certificates in use by your organization; the exact number of CA certificates will depend on the structure of your organization's PKI.
+
+### Verify the Puppet master host certificate against the CA bundle
+
+Before installing the new certificate bundle, verify that the bundle can be used to verify the existing host certificate on the CA server. If this step fails then the CA certificate bundle is invalid and cannot be installed.
+
+    openssl verify -CAfile ca-bundle.pem $(puppet master --configprint hostcert)
+
+### Update the Puppet CA certificate bundle
+
+    cp ca-bundle.pem /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem
+
+### Update the Puppet agent CA certificate bundle
+
+The Puppet agent caches the CA certificate and will not automatically fetch the new CA certificate bundle from the master. This caveat applies to the Puppet CA server itself as well. On the CA server, copy the CA certificate bundle to the local CA certificate path:
+
+    cp /etc/puppetlabs/puppet/ssl/ca/ca_crt.pem /etc/puppetlabs/puppet/ssl/certs/ca.pem
+
+If any Puppet agents have already been created, copy the CA certificate bundle to `/etc/puppetlabs/puppet/ssl/certs/ca.pem` for every Puppet agent that was created before the new CA certificate was generated.
+
+Note that due to [PUP-6697](https://tickets.puppetlabs.com/browse/PUP-6697) Puppet agents are unable to download CA certificate bundles and will only save the first certificate in the bundle. Since the first certificate in the bundle will be the Puppet intermediate CA certificate, if the Puppet agent downloads and saves this single CA certificate it will not be able to complete verification of the Puppet Server certificate and will fail to connect.
+
+There may be other settings in the `webserver.conf` file; the other settings should be left alone.
+
+### Restart Puppet Server
+
+Once the new CA certificate is installed, restart pe-puppetserver.
```

Comment:
Should this note puppetserver in addition to pe-puppetserver?

Is a full puppetserver service restart required?